src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -80,28 +82,34 @@ def get_dxlink_ids(link):
     '''
     :param link: A DNAnexus link
     :type link: dict
-    :returns: Object ID, Project ID (or :const:`None` if no project specified in the link)
+    :returns: (Object ID, Project ID) if the link is to a data object (or :const:`None`
+        if no project specified in the link), or (Job ID, Field) if the link is
+        a job-based object reference.
     :rtype: tuple
 
     Returns the object and project IDs stored in the given DNAnexus
     link.
     '''
-    if isinstance(link['$dnanexus_link'], dict):
+    if not is_dxlink(link):
+        raise DXError('Invalid link: %r' % link)
+    if isinstance(link['$dnanexus_link'], basestring):
+        return link['$dnanexus_link'], None
+    elif 'id' in link['$dnanexus_link']:
         return link['$dnanexus_link']['id'], link['$dnanexus_link'].get('project')
     else:
-        return link['$dnanexus_link'], None
+        return link['$dnanexus_link']['job'], link['$dnanexus_link']['field']
 
 def _guess_link_target_type(link):
     if is_dxlink(link):
-        # Guaranteed by is_dxlink that one of the following will work
-        if isinstance(link['$dnanexus_link'], basestring):
-            link = link['$dnanexus_link']
-        else:
-            link = link['$dnanexus_link']['id']
-    class_name, _id = link.split("-", 1)
-    class_name = 'DX'+class_name.capitalize()
-    if class_name == 'DXGlobalworkflow':
-        class_name = 'DXGlobalWorkflow'
+        link = get_dxlink_ids(link)[0]
+    class_name = 'DX' + link.split("-", 1)[0].capitalize()
```

Comment:
If I understand the code right, the following comment would be helpful:
// Now we have an object-id, for example: "file-xxxx", "record-xxxx", ...